[i3ass][i3fyra][i3viswiz]  i3viswiz accepts any amount of arguments as long as the last one starts with `u|l|d|r|up|left|down|right`

**Description**  
- 2) I found out that i3viswiz does accept some (I suppose) non-intended input. It seems i3viswiz will accept any amount of arguments, as long as the last argument is any of `u|l|d|r|up|left|down|right`. 

**To Reproduce**  
Steps to reproduce the behavior:  
- see terminal commands and output below:


as intended, direction only:  
```bash
~ > i3viswiz --verbose l
m [con_mark=i3viswiz-last-direction=94457275819920] mark --toggle --add i3viswiz-last-direction=94457275819920
m [con_id=94457275831344] mark --add i3viswiz-last-direction=94457275819920
m [con_id=94457275960096] focus
[{"success":true},{"success":true},{"success":true}]
```

single extra argument before direction:  
```bash
~ > i3viswiz --verbose asdfasdf l  
m [con_mark=i3viswiz-last-direction=94457275819920] mark --toggle --add i3viswiz-last-direction=94457275819920
m [con_id=94457275831344] mark --add i3viswiz-last-direction=94457275819920
m [con_id=94457275960096] focus
[{"success":true},{"success":true},{"success":true}]
```

multiple extra arguments before direction:  
```bash
~ > i3viswiz --verbose asdfasdf qwer asdfgasdf l
m [con_mark=i3viswiz-last-direction=94457275819920] mark --toggle --add i3viswiz-last-direction=94457275819920
m [con_id=94457275831344] mark --add i3viswiz-last-direction=94457275819920
m [con_id=94457275960096] focus
[{"success":true},{"success":true},{"success":true}]
```

multiple extra arguments, direction inbetween:  
```bash
~ > i3viswiz --verbose blah bar left nei
[ERROR] nei not valid direction (l|r|u|d)
```

multiple extra arguments, last one starting with direction:
```bash
~ > i3viswiz --verbose blah bar rleftnei 
m [con_mark=i3viswiz-last-direction=94457276140256] mark --toggle --add i3viswiz-last-direction=94457276140256
m [con_id=94457275831344] mark --add i3viswiz-last-direction=94457275819920
m [con_id=94457275960096] focus
[{"success":true},{"success":true},{"success":true}]
```

Directionality doesn't matter here, i3viswiz chooses whatever valid argument it finds first, so `i3viswiz blah barnei lright` will focus to the left.

**Expected behavior**  
I was expecting an error message on multiple arguments and/or all arguments starting with `u|l|d|r|up|left|down|right`


**Desktop (please complete the following information):**  
- Arch btw

```bash
~ > i3 --version
i3 version 4.22 (2023-01-02) © 2009 Michael Stapelberg and contributors
```

Yeah, this was smoother than #196 , i will merge #197 . Below is new error messages:  

```
$ i3viswiz --verbose asdfasdf l  
[ERROR] i3viswiz: direction ('asdfasdf') is not valid! ( left|right|up|down|l|r|u|d )

$ i3viswiz --verbose asdfasdf qwer asdfgasdf l
[ERROR] i3viswiz: direction ('asdfasdf') is not valid! ( left|right|up|down|l|r|u|d )

$ i3viswiz --verbose blah bar left nei
[ERROR] i3viswiz: direction ('blah') is not valid! ( left|right|up|down|l|r|u|d )

$ i3viswiz --verbose left right up
[ERROR] i3viswiz: expected 0 arguments got '(right up)'

```

--- a/src/i3viswiz/config.mak
+++ b/src/i3viswiz/config.mak
@@ -1,7 +1,7 @@
 NAME         := i3viswiz
-VERSION      := 0.55
+VERSION      := 0.56
 CREATED      := 2018-01-18
-UPDATED      := 2022-05-21
+UPDATED      := 2023-07-22
 AUTHOR       := budRich
 CONTACT      := https://github.com/budlabs/i3ass
 USAGE        := options

--- a/src/i3viswiz/i3viswiz
+++ b/src/i3viswiz/i3viswiz
@@ -2,10 +2,14 @@
 
 main(){
 
-  (($#)) && {
-    arg_direction=${!#:0:1} arg_direction=${arg_direction,,}
-    [[ $arg_direction =~ l|r|u|d ]] \
-      || ERX "${!#} not valid direction (l|r|u|d)"
+  [[ ${arg_direction:=${1,,}} ]] && { 
+    [[ ${arg_direction} =~ ^(l|r|u|d|left|right|up|down)$ ]]  \
+      || ERX "i3viswiz: direction ('$arg_direction') is not valid!" \
+             "( left|right|up|down|l|r|u|d )"
+
+    shift
+    (($#)) && ERX "i3viswiz: expected 0 arguments got '($*)'"
+    arg_direction=${arg_direction:0:1}
   }
 
   : "${_o[json]:=$(i3-msg -t get_tree)}"

